powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v2/TokenBehavior.java

```diff
@@ -172,8 +169,9 @@ private EciesPayload createToken(String activationId, String ephemeralPublicKeyB
             final ObjectMapper mapper = new ObjectMapper();
             final byte[] tokenBytes = mapper.writeValueAsBytes(tokenInfo);
 
-            final BasicEciesDecryptor decryptor = new BasicEciesDecryptor((ECPrivateKey) privateKey);
-            return decryptor.encrypt(tokenBytes, (ECPublicKey) ephemeralPublicKey, ephemeralPublicKeyBytes);
+            final EciesDecryptor decryptor = new EciesDecryptor((ECPrivateKey) privateKey);
+            // The encryptResponseDirect() method is only used in version 2.0, because there is no incoming data to decrypt.
+            return decryptor.encryptResponseDirect(tokenBytes, ephemeralPublicKeyBytes);
```

Comment:
OK - I was a bit suspicious about this code before, but we should rewrite this. It is confusing - we work with key and then we randomly use `byte[]` to pass the public key, that is internally converted to key as a side effect and used as `sharedInfo1`. We should not need the`encryptResponseDirect` method at all if we just follow the ECIES template with the private key, public key and correctly setup `sharedInfo1` attribute (there is no change in functionality, we only updated API)... The problem propagates all the way to the `EnvelopeKey` class in crypto...

Reply:
I agree the `encryptResponseDirect` method is unusual. Note that it is only used for V2 tokens where there is no incoming `encryptedData` and `mac`. The method is equivalent to the original method:
https://github.com/wultra/powerauth-crypto/blob/2b420826bcf03e300ec72748818bf01d3a17d9cd/powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/BasicEciesDecryptor.java#L176

I will look into rewriting the method.

Reply:
Let's have a look tomorrow.

Reply:
We agreed to keep the method as is for now and deprecate it as soon as version 2.0 of crypto is no longer used. We will only use the method once for V2 tokens.